.github/workflows/ci.yml

```diff
@@ -212,23 +206,51 @@ jobs:
         name: packages-v${{ steps.versions.outputs.product_full_version }}
         if-no-files-found: error
         path: "GeneratedNuGetPackages/${{ steps.versions.outputs.product_configuration }}/*.*nupkg"
+    - name: Upload SystemTests
+      uses: actions/upload-artifact@v4
+      with:
+        name: SystemTests
+        if-no-files-found: error
+        path: Tests/Reqnroll.SystemTests/bin/
+    - name: Upload Specs
+      uses: actions/upload-artifact@v4
+      with:
+        name: Specs
+        if-no-files-found: error
+        path: Tests/Reqnroll.Specs/bin/
 
   specs:
     runs-on: ubuntu-latest
     needs: build
+    env:
+      # Disable telemetry, because we use many small projects
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+      REQNROLL_TELEMETRY_ENABLED: 0
 
     steps:
     - uses: actions/checkout@v4
-    - name: Restore dependencies
-      run: dotnet restore
-    - name: Install Test Report Dependencies
-      run: | 
-        dotnet add ./Tests/Reqnroll.Specs/Reqnroll.Specs.csproj package GitHubActionsTestLogger
-    - name: Build
-      run: dotnet build --no-restore ${{ needs.build.outputs.build_params }}
+    - name: Set Runner ImageVersion to Env
+      shell: pwsh
+      run: |
+        Write-Output "ImageVersion=$env:ImageVersion" >> $env:GITHUB_ENV
+    - name: Cache RRC folder
+      uses: actions/cache@v4
+      with:
+        path: /tmp/RRC
+        key: RRC-Specs-${{ runner.os }}-${{ env.ImageVersion }} # Use the runner version as key, so if new dependencies get installed we get a new cache entry
+    - name: Download packages
+      uses: actions/download-artifact@v4
+      with:
+        name: packages-v${{ needs.build.outputs.product_full_version }}
+        path: GeneratedNuGetPackages/${{ needs.build.outputs.product_configuration }}/
+    - name: Download Specs
+      uses: actions/download-artifact@v4
+      with:
+        name: Specs
+        path: Tests/Reqnroll.Specs/bin/
     - name: Specs
       shell: pwsh
-      run: dotnet test ./Tests/Reqnroll.Specs/Reqnroll.Specs.csproj --filter "Category!=quarantaine{{ needs.build.outputs.specs_filter }}" --logger "trx;LogFileName=${{ github.workspace }}/TestResults/specs-results.trx" ${{ needs.build.outputs.test_params }}
+      run: dotnet test ./Tests/Reqnroll.Specs/bin/${{ needs.build.outputs.product_configuration }}/*/Reqnroll.Specs.dll --filter "Category!=quarantaine{{ needs.build.outputs.specs_filter }}" --logger "trx;LogFileName=${{ github.workspace }}/TestResults/specs-results.trx" ${{ needs.build.outputs.test_params }}
```

Comment:
Would you mind explaining what this change from running the "project" to running (multiple?) dlls achieves?

Reply:
This allows us to avoid compiling the test assemblies in different jobs and also avoids compiling Reqnroll itself multiple times (Reqnroll depends on the project reference via Specs).
This improves CI/CD execution time by avoiding the need to call dotnet new sln/project.
More importantly, this ensures that we use the compiled assemblies from the "Build" job, which is published, in all our tests.